src/adsp_config.c

```diff
@@ -1773,6 +1773,9 @@ static int parse_pin(const toml_table_t *mod_entry, struct parse_ctx *ctx,
 	ext_mod_config->pin_desc = calloc(sizeof(const struct fw_pin_description),
 					  toml_array_nelem(arr) / 6);
 
+	if(!ext_mod_config->pin_desc)
+		return err_key_parse("pin", "failed to allocate description");
+
```

Comment:
Please try `err_malloc()` instead of `err_key_parse()`

Reply:
Like that?

Reply:
great!